src/Nethermind/Nethermind.Consensus.AuRa/AuRaBlockProcessor.cs

```diff
@@ -113,10 +116,33 @@ private void ValidateGasLimit(BlockHeader header, BlockHeader parentHeader)
 
         private void ValidateTxs(Block block, BlockHeader parentHeader)
         {
+            (bool Allowed, string Reason)? TryRecoverSenderAddress(Transaction tx)
+            {
+                if (tx.Signature != null)
+                {
+                    IReleaseSpec spec = _specProvider.GetSpec(block.Number);
+                    var ecdsa = new EthereumEcdsa(_specProvider.ChainId, LimboLogs.Instance);
+                    Address txSenderAddress = ecdsa.RecoverAddress(tx, !spec.ValidateChainId);
+                    if (tx.SenderAddress != txSenderAddress)
+                    {
+                        if (_logger.IsWarn) _logger.Warn($"Transaction {tx.ToShortString()} in block {block.ToString(Block.Format.FullHashAndNumber)} had recovered sender address on validation.");
+                        tx.SenderAddress = txSenderAddress;
+                        return _txFilter.IsAllowed(tx, parentHeader);
+                    }
+                }
+
+                return null;
+            }
+
             for (int i = 0; i < block.Transactions.Length; i++)
             {
                 var tx = block.Transactions[i];
                 var txFilterResult = _txFilter.IsAllowed(tx, parentHeader);
+                if (!txFilterResult.Allowed)
+                {
```

Comment:
only call once again when recovery returned different address than before, otherwise you have a huge performance hit an can be the target of attack on tx pool

Reply:
ah no, this is processor